akka-typed/src/main/scala/akka/typed/internal/ActorCell.scala

```diff
@@ -326,6 +315,23 @@ private[typed] class ActorCell[T](
         else unsafe.getAndAddInt(this, statusOffset, -1)
       }
     }
+
+    try {
+      unscheduleReceiveTimeout()
+      if (!isTerminated(status))
+        process()
+      scheduleReceiveTimeout()
+      afterProcess()
+    } catch {
+      case NonFatal(ex) ⇒
+        fail(ex)
+        afterProcess()
```

Comment:
I see; then how about moving the re-execution out of the block and returning a boolean from afterProcess that signals whether to enqueue? This way we ensure that if something is caught afterProcess cannot have been started, because it cannot reasonably throw.